### PR TITLE
Fixes #7: Scaladoc Improvements

### LINKED ISSUE #7: Convert scaladoc from markdown to scaladoc syntax
I can never remember scaladoc's markup format, so I generally just write markdown and fix it later.  Now is later!

### PR DESCRIPTION
**Pull request by alexandru**
Related to https://github.com/typelevel/cats-effect/issues/7

### CODE REVIEW

File: core/shared/src/main/scala/cats/effect/Effect.scala [hunk 1]
```diff
@@ -22,13 +22,17 @@ import simulacrum._
 import scala.concurrent.ExecutionContext
 import scala.util.Either
 
+/**
+ * A monad that can suspend side-effects in the `F` context and that
+ * supports lazy and potentially asynchronous evaluation.
+ */
```
Comment (rossabaker) [inline core/shared/src/main/scala/cats/effect/Effect.scala]: Style nitpick: `Async` is "A Monad for `F` datatypes" vs. `Effect` is "A monad ... in the `F` context".  Both wordings are fine, but I'd pick one to give it a single voice.

No hyphen in "side-effects"

File: core/shared/src/main/scala/cats/effect/Sync.scala [hunk 2]
```diff
@@ -19,8 +19,27 @@ package effect
...
+/**
+ * A monad that can suspend the execution of side-effects
+ * in the `F[_]` context.
+ */
 @typeclass
 trait Sync[F[_]] extends MonadError[F, Throwable] {
+  /**
+   * Suspends the evaluation of an `F` reference.
+   *
+   * Equivalent with `FlatMap.flatten` for pure expressions,
```
Comment (rossabaker) [inline core/shared/src/main/scala/cats/effect/Sync.scala]: s/Equivalent with/Equivalent to/

File: core/shared/src/main/scala/cats/effect/Sync.scala [hunk 3]
```diff
@@ -19,8 +19,27 @@ package effect
...
+   * the purpose of this function is to suspend side-effects
+   * in `F`.
+   */
   def suspend[A](thunk: => F[A]): F[A]
+
+  /**
+   * Lifts any by-name parameter in the `F` context.
```
Comment (rossabaker) [inline core/shared/src/main/scala/cats/effect/Sync.scala]: s/in/into/

**Comment by alexandru** [comment]
Thanks for the review,

@djspiewak I've reverted the changes to `build.sbt` and addressed @rossabaker's comments.

**Pull request merged**